tdp/cli/commands/deploy.py

```diff
@@ -86,72 +38,57 @@ def deploy(
     vars,
 ):
     if not vars.exists():
-        raise click.BadParameter(f"{vars} does not exist")
-    dag = Dag(collections)
-    set_nodes = set()
-    if sources:
-        sources = sources.split(",")
-        set_nodes.update(sources)
-    if targets:
-        targets = targets.split(",")
-        set_nodes.update(targets)
-    set_difference = set_nodes.difference(dag.operations)
-    if set_difference:
-        raise click.BadParameter(f"{set_difference} are not valid nodes")
-    run_directory = run_directory.absolute() if run_directory else None
+        raise click.BadParameter(f"{vars} does not exist.")
+
+    cluster_variables = ClusterVariables.get_cluster_variables(
+        collections, vars, validate=validate
+    )
+    check_services_cleanliness(cluster_variables)
 
-    ansible_executor = AnsibleExecutor(
-        run_directory=run_directory,
-        dry=dry or mock_deploy,
+    deployment_runner = DeploymentRunner(
+        collections,
+        AnsibleExecutor(
+            run_directory=run_directory.absolute() if run_directory else None,
+            dry=dry or mock_deploy,
+        ),
+        cluster_variables,
     )
+
     session_class = get_session_class(database_dsn)
     with session_class() as session:
-        cluster_variables = ClusterVariables.get_cluster_variables(
-            collections, vars, validate=validate
-        )
-        check_services_cleanliness(cluster_variables)
-
-        deployment_runner = DeploymentRunner(
-            collections, ansible_executor, cluster_variables
-        )
-        if sources:
-            click.echo(f"Deploying from {sources}")
-        elif targets:
-            click.echo(f"Deploying to {targets}")
-        else:
-            click.echo(f"Deploying TDP")
-        try:
-            deployment_plan = DeploymentPlan.from_dag(
-                dag,
-                sources=sources,
-                targets=targets,
-                filter_expression=filter,
-                filter_type=filter_type,
-                restart=restart,
+        planned_deployment_log = get_planned_deployment_log(session)
+        if planned_deployment_log is None:
+            raise click.ClickException(
+                "No previous deployment found, please run `tdp plan` first."
```

Comment:
Also need to be changed here.
```suggestion
                "No planned deployment found, please run `tdp plan` first."
```

Reply:
Done